pkg/apis/traceflow/v1/types.go

```diff
@@ -0,0 +1,162 @@
+	FORWARDING    Component = "FORWARDING"
+)
+
+type Action string
+
+const (
+	DELIVERED Action = "DELIVERED"
+	RECEIVED  Action = "RECEIVED"
+	FORWARDED Action = "FORWARDED"
+	DROPPED   Action = "DROPPED"
+)
+
+type PacketResouceType string
+
+const (
+	FieldsPacketData PacketResouceType = "FieldsPacketData"
+)
+
+type PacketTransportType string
+
+const (
+	UNICAST PacketTransportType = "UNICAST"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type Traceflow struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Source      `json:"source"`
+	Destination `json:"destination"`
+	RoundID     string `json:"roundID,omitempty"`
+
+	Packet `json:"packet"`
+	Status `json:"status"`
+}
+
+type Source struct {
+	SrcNamespace string `json:"srcNamespace,omitempty"`
+	SrcPod       string `json:"srcPod,omitempty"`
+}
+
+type Destination struct {
+	DstNamespace string `json:"dstNamespace,omitempty"`
+	DstPod       string `json:"dstPod,omitempty"`
+	DstService   string `json:"dstService,omitempty"`
+	DstIP        string `json:"dstIP,omitempty"`
+}
+
+type IPHeader struct {
+	SrcIP    string `json:"srcIP,omitempty"`
+	Protocol int32  `json:"protocol,omitempty"`
+	TTL      int32  `json:"ttl,omitempty"`
+	Flags    int32  `json:"flags,omitempty"`
+}
+
+type TransportHeader struct {
+	ICMPEchoRequestHeader `json:"icmpEchoRequestHeader"`
+	UDPHeader             `json:"udpHeader"`
+	TCPHeader             `json:"tcpHeader"`
+}
+
+type ICMPEchoRequestHeader struct {
+	ID       int32 `json:"id,omitempty"`
+	Sequence int32 `json:"sequence,omitempty"`
+}
+
+type UDPHeader struct {
+	SrcUDPPort int32 `json:"srcUDPPort,omitempty"`
+	DstUDPPort int32 `json:"dstUDPPort,omitempty"`
+}
+
+type TCPHeader struct {
+	SrcTCPPort int32 `json:"srcTCPPort,omitempty"`
+	DstTCPPort int32 `json:"dstTCPPort,omitempty"`
+	TCPFlags   int32 `json:"tcpFlags,omitempty"`
+}
+
+type Packet struct {
+	ResouceType   PacketResouceType   `json:"resouceType,omitempty"`
+	TransportType PacketTransportType `json:"transportType,omitempty"`
+	PayloadString string              `json:"payloadString,omitempty"`
+	PayloadSize   int32               `json:"payloadSize,omitempty"`
+
+	IPHeader        `json:"ipHeader"`
+	TransportHeader `json:"transportHeader"`
+}
+
+type Status struct {
+	Phase        Phase         `json:"phase,omitempty"`
+	CrossNodeTag uint8         `json:"crossNodeTag,omitempty"`
+	NodeSender   []Observation `json:"nodeSender,omitempty"`
+	NodeReceiver []Observation `json:"nodeReceiver,omitempty"`
+}
+
+type Observation struct {
+	Component       Component  `json:"component,omitempty"`
+	SubComponent    string     `json:"subComponent,omitempty"`
+	ComponentName   string     `json:"componentName,omitempty"`
+	Action          Action     `json:"action,omitempty"`
+	RoundID         string     `json:"roundID,omitempty"`
+	NodeUUID        string     `json:"nodeUUID,omitempty"`
```

Comment:
As we assume users can look at the CRD directly, I think we should use user readable names, so I would use NodeName, PodNamespace/Name, instead of UIDs.

Reply:
Yes, we will add the name, but I prefer to keep the UID.

Reply:
Do you have any specific use case for UID? In K8s, clients typically use name or namespace + name.
And for end user, I do not think UID is convenient.

Reply:
OK. Let's use Node instead of NodeUUID.